### ISSUE TITLE
unbound variable in query (bug in 0.37.6 and after)

### ISSUE THREAD

**Issue by stephenctw** [dolthub/dolt#3100]
There's a bug after 0.37.5 (exclusive) with the variables bindings.

```
Starting server with Config HP="localhost:3306"|U="root"|P=""|T="28800000"|R="false"|L="info"
2022-03-28T13:45:09+08:00 INFO [conn 1] NewConnection {DisableClientMultiStatements=false}
2022-03-28T13:45:09+08:00 WARN [conn 1] error running query {connectTime=2022-03-28T13:45:09+08:00, error=unbound variable "v5" in query, query=insert into nodes(id,owner,status,timestamp) values(?,?,?,?) on duplicate key update owner=?,status=?,timestamp=?}
2022-03-28T13:45:09+08:00 INFO [conn 1] ConnectionClosed {}
```

Here's the the database schema and query that throws me an error in 0.37.6 and 0.37.7.

schema:
```sql
CREATE TABLE `users` (
  `id` varchar(42) NOT NULL PRIMARY KEY,
  `stakedBalance` numeric(30,0) NOT NULL DEFAULT 0,
  `maturingBalance` numeric(30,0) NOT NULL DEFAULT 0,
  `releasingBalance` numeric(30,0) NOT NULL DEFAULT 0,
  `maturingTimestamp` bigint NOT NULL DEFAULT 0,
  `releasingTimestamp` bigint NOT NULL DEFAULT 0,
  `balance` numeric(30,0) NOT NULL DEFAULT 0,
  `timestamp` bigint NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

CREATE TABLE `nodes` (
  `id` varchar(42) NOT NULL PRIMARY KEY,
  `owner` varchar(42),
  `status` varchar(12),
  `timestamp` bigint NOT NULL,
  FOREIGN KEY(`owner`) REFERENCES users(`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

CREATE TABLE `protocols` (
  `id` varchar(42) NOT NULL PRIMARY KEY,
  `version` int NOT NULL,
  `timestamp` int NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

CREATE TABLE `chains` (
  `protocol` varchar(42) NOT NULL,
  `number` int NOT NULL,
  `targetInterval` numeric(30,0) NOT NULL,
  `timestamp` bigint NOT NULL,
  PRIMARY KEY(`protocol`,`number`),
  FOREIGN KEY(`protocol`) REFERENCES protocols(id)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

CREATE TABLE `blocks` (
  `id` varchar(66) NOT NULL PRIMARY KEY,
  `protocol` varchar(42) NOT NULL,
  `chain` int NOT NULL,
  `number` int NOT NULL,
  `producer` varchar(42) NOT NULL,
  `node` varchar(42) NOT NULL,
  `reward` numeric(30,0),
  `difficulty` numeric(30,0) NOT NULL,
  `gasPrice` numeric(30,0) NOT NULL DEFAULT 0,
  `gasLimit` numeric(30,0) NOT NULL DEFAULT 0,
  `timestamp` bigint NOT NULL,
  FOREIGN KEY(`protocol`,`chain`) REFERENCES chains(`protocol`,`number`),
  FOREIGN KEY(`producer`) REFERENCES users(`id`),
  FOREIGN KEY(`node`) REFERENCES nodes(`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
```
reproducible code:
```rs
use diesel::connection::Connection;
use diesel::mysql::MysqlConnection;
use diesel::sql_types::{BigInt, Nullable, VarChar};
use diesel::{sql_query, RunQueryDsl};
use ethers::types::{Address, H256, U256, U64};

#[tokio::main]
async fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
    let sql_connection = MysqlConnection::establish("mysql://root@127.0.0.1:3306/dolt_pos")?;

    let upsert_query = sql_query("insert ignore into users(id,timestamp) values (?,?)")
        .bind::<VarChar, _>("0x64a7252bad0cc874da46935c80cf84b69a6764c3".to_string())
        .bind::<BigInt, _>(1608041846);

    upsert_query.execute(&sql_connection)?;

    let upsert_query = sql_query(
        "insert into nodes(id,owner,status,timestamp) values(?,?,?,?) on duplicate key update owner=?,status=?,timestamp=?",
    )
    // insert new row
    .bind::<VarChar, _>("0x6eb65260d044f4e72696398e2f122db9989c3de8".to_string())
    .bind::<Nullable<VarChar>, _>(Some("0x64a7252bad0cc874da46935c80cf84b69a6764c3".to_string()))
    .bind::<Nullable<VarChar>, _>(Some("Owned".to_string()))
    .bind::<BigInt, _>(1608041846)
    // update on existing row
    .bind::<Nullable<VarChar>, _>(Some("0x64a7252bad0cc874da46935c80cf84b69a6764c3".to_string()))
    .bind::<Nullable<VarChar>, _>(Some("Owned".to_string()))
    .bind::<BigInt, _>(1608041846);

    upsert_query.execute(&sql_connection)?;

    Ok(())
}

```

The code to reproduce this bug is written in Rust, should be easy to rewrite in other scripting languages.

**Comment by timsehn** [comment]
@max-hoffman has been changing this code. I'm not sure hos changes made it out but th9s is at least a good test to add.

**Comment by max-hoffman** [comment]
@stephenctw thanks for the bug report! Looking at this today.

**Comment by zachmu** [comment]
I believe this is fixed in source but unreleased, right Max? Can we do a release and close this out?

**Comment by max-hoffman** [comment]
@stephenctw main does not have fixes for this yet, but I think I'll be able to merge and release a PR that does by tomorrow. I'll keep you updated on my status.

**Comment by max-hoffman** [comment]
The bigger PR got delayed but I added a fix specifically for this bug this morning https://github.com/dolthub/go-mysql-server/pull/924.

**Comment by max-hoffman** [comment]
This should be fixed in dolt `0.37.9`, released last Friday. If you find similar bugs please follow up!

**Comment by stephenctw** [comment]
> This should be fixed in dolt `0.37.9`, released last Friday. If you find similar bugs please follow up!

Confirmed the bug is fixed in `0.37.9`, thanks for the support!
